pkg/monitor/monitor/monitor_manager.go

```diff
@@ -84,6 +88,34 @@ func (mm *MonitorManager) SyncMonitor(monitor *v1alpha1.TidbMonitor) error {
 	if monitor.DeletionTimestamp != nil {
 		return nil
 	}
+	if monitor.Spec.Clusters == nil || len(monitor.Spec.Clusters) < 1 {
+		err := fmt.Errorf("tm[%s/%s] does not configure the target tidbcluster", monitor.Namespace, monitor.Name)
+		return err
+	}
+	tcRef := monitor.Spec.Clusters[0]
+	if len(tcRef.Namespace) < 1 {
+		tcRef.Namespace = monitor.Namespace
+	}
+	tc, err := mm.cli.PingcapV1alpha1().TidbClusters(tcRef.Namespace).Get(tcRef.Name, metav1.GetOptions{})
+	if err != nil {
+		rerr := fmt.Errorf("get tm[%s/%s]'s target tc[%s/%s] failed, err: %v", monitor.Namespace, monitor.Name, tcRef.Namespace, tcRef.Name, err)
+		return rerr
+	}
+	if tc.Status.Monitor != nil {
+		if tc.Status.Monitor.Name != monitor.Name || tc.Status.Monitor.Namespace != monitor.Namespace {
+			err := fmt.Errorf("tm[%s/%s]'s target tc[%s/%s] already referenced by TidbMonitor [%s/%s]", monitor.Namespace, monitor.Name, tc.Namespace, tc.Name, tc.Status.Monitor.Namespace, tc.Status.Monitor.Name)
+			mm.recorder.Event(monitor, corev1.EventTypeWarning, FailedSync, err.Error())
+			return err
+		}
+	}
+
+	// TODO: support Validating hook for updating tidbcluster that forbid update monitorRef with a different reference
+	// Patch tidbcluster status first in order to avoiding let multi tidbmonitor monitor the same tidbcluster
+	if err := mm.patchTidbClusterStatus(&tcRef, monitor); err != nil {
+		message := fmt.Sprintf("Sync TidbMonitorRef into targetCluster[%s/%s] status failed, err:%v", tc.Namespace, tc.Name, err)
+		mm.recorder.Event(monitor, corev1.EventTypeWarning, FailedSync, message)
+		return err
```

Comment:
Is it OK that we just emit event and print error log here and continue to sync the resources without returning error?

Reply:
I think we should move the status update code block to the end of the sync function, we should update the resource first then update its reference.

Reply:
I think the monitorRef in Tidbcluster Ref could also seem as a mutex contention that tidbmonitor only success syncing its own information to the target tidbcluster status, then the tidbmonitor would start to sync the resources. That is to say, tidbmonitor would try to patch the tidbcluster first.

Reply:
> Is it OK that we just emit event and print error log here and continue to sync the resources without returning error?

updated, using log instead of event.